src/item.cpp

```diff
@@ -94,6 +94,22 @@ item::item(const std::string new_type, unsigned int turn, bool rand, const hande
     // TODO: some item types use the same member (e.g. charges) for different things. Handle or forbid this.
     if( type->gun ) {
         charges = 0;
+        if( !type->gun->built_in_mods.empty() ){
```

Comment:
And actually (sorry, I forgot to mention that earlier), you don't need to check for the vector being empty here, or *anywhere* before looping over its content. If it's empty, the loop will be skipped anyway. Checking for emptiness is completely pointless.

Same in the loop below and in item_factory.cpp